bolt/test/lsda.cpp

```diff
@@ -0,0 +1,47 @@
+// This test check that LSDA section named by .gcc_except_table.main is
+// disassembled by BOLT.
+
+// RUN: %clang++ %cxxflags -O3 -flto=thin -no-pie -c %s -o %t.o
+// RUN: %clang++ %cxxflags -flto=thin -no-pie -fuse-ld=lld %t.o -o %t.exe \
```

Comment:
Is ThinLTO flag needed for this test case?

Reply:
Currently definitely not since I've added @rafaelauler offer to the ld script at least (I'm not sure does the problem has any connection to lto by default, probably not). I've just re-used the test provided in #71804 . Does it raise some problem here?

Reply:
I’ve had an issue on a system where lld’s version didn’t match that of the compiler. Uncommon scenario, but nevertheless it’s better to fix.